### PR TITLE
re-enable OpenMP in wheels

### LINKED ISSUE #3400: Reenable OpenMP threading backend in wheel builds
- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).

On Linux:
* The 0.40.0 wheel build of Numba had an OpenMP backend that segfaulted/sigabrt'd `ld.so`.
* The 0.40.1 wheel build of Numba disabled OpenMP to avoid the above problem.

This issue is to act as a reminder to debug and fix the root cause of the problem, reproducer etc. is here https://github.com/numba/numba/issues/3341

### PR DESCRIPTION
**Pull request by esc**
Fixes #3400 - now that `patchelf` is working again we can go ahead and
re-enable OpenMP in the wheel build.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -238,9 +238,8 @@ def check_file_at_path(path2file):
     else:
         print("TBB not found")
 
-    # Disable OpenMP if we are building a wheel or
-    # forced by user with NUMBA_NO_OPENMP=1
-    if is_building_wheel() or os.getenv('NUMBA_NO_OPENMP'):
```
Comment (stuartarchibald) [inline setup.py:243]: is `is_building_wheel` now dead?

Comment (esc) [reply setup.py:243]: No, it is still used on line 29 in that file.

Comment (stuartarchibald) [reply setup.py:243]: Are you sure? https://github.com/numba/numba/pull/5944/files#diff-2eeaed663bd0d25b7e608891384b7298R29

Comment (esc) [reply setup.py:243]: Oh,my bad, that is the definition...

**Comment by esc** [comment]
Note to self: this needs to be tested manually by building manylinux wheels.

**Comment by sklam** [comment]
needs to test it on manylinux*

**Comment by esc** [comment]
Build-farm ID: `numba_wheel_20`.

**Comment by esc** [comment]
These are the results from running the code that checks that `patchelf` (from #3400) has done it's job correctly:

Screen Shot 2020-07-21 at 11 43 10

**Comment by stuartarchibald** [comment]
I've manually checked one of the wheel samples:

```
$ wget https://anaconda.org/sklam/numba/0.51.0.dev0+124.g9e4905650.dirty/download/numba-0.51.0.dev0+124.g9e4905650.dirty-cp37-cp37m-manylinux2014_x86_64.whl

$ unzip numba-0.51.0.dev0+124.g9e4905650.dirty-cp37-cp37m-manylinux2014_x86_64.whl

$ cd numba

$ ldd ./numba/np/ufunc/omppool.cpython-37m-x86_64-linux-gnu.so
        <snip>
        libgomp-7c85b1e2.so.1.0.0 => /tmp/ompwheel/./numba/np/ufunc/../../.libs/libgomp-7c85b1e2.so.1.0.0 (0x00007efe0b5a1000)
        <snip>

$ function dso_info(){
> printf "Info for bad DSO:\n"
> printf "===>OMP symbols in DT_NEEDED:"
> readelf -d $OFFENDING_DSO|grep omp
> printf "===>OMP symbols in Versioned:"
> readelf -WV $OFFENDING_DSO|grep omp
> printf "===>ELF linter output:"
> eu-elflint $OFFENDING_DSO
> printf "\n"
> }

$ OFFENDING_DSO=./np/ufunc/omppool.cpython-37m-x86_64-linux-gnu.so

$ dso_info 
Info for bad DSO:
===>OMP symbols in DT_NEEDED: 0x0000000000000001 (NEEDED)             Shared library: [libgomp-7c85b1e2.so.1.0.0]
===>OMP symbols in Versioned:  0x0090: Version: 1  File: libgomp-7c85b1e2.so.1.0.0  Cnt: 2
===>ELF linter output:phdr[2]: unknown object file note type 425984 with owner name '' at offset 24
phdr[2]: extra 12 bytes after last note
section [15] '.tbss': thread-local data sections address not zero
section [30] '.symtab': _GLOBAL_OFFSET_TABLE_ symbol size 0 does not match .got.plt section size 400
section [30] '.symtab': symbol 76: st_value out of bounds
section [30] '.symtab': _DYNAMIC_ symbol value 0x206d90 does not match dynamic segment address 0x209000
section [30] '.symtab': _DYNAMIC symbol size 0 does not match dynamic segment size 560
section [34] '.note.gnu.build-id': cannot get content of note section
loadable segment [7] is writable but contains no writable sections

$ conda create -n _tmp_wheel_omp python=3.7 numpy=1.18

$ conda activate _tmp_wheel_omp37

$ pip install numba-0.51.0.dev0+124.g9e4905650.dirty-cp37-cp37m-manylinux2014_x86_64.whl

$ cat t1.py
from numba import njit, prange
import numpy as np

@njit(parallel=True)
def foo(n):
    acc = 0
    for i in prange(n):
        acc += 10
    return acc

foo(100)

$ LD_DEBUG=libs python t1.py 2>&1|grep libgomp
     15433:     find library=libgomp-7c85b1e2.so.1.0.0 [0]; searching
     15433:       trying file=/tmp/ompwheel/numba/np/ufunc/../../.libs/tls/x86_64/libgomp-7c85b1e2.so.1.0.0
     15433:       trying file=/tmp/ompwheel/numba/np/ufunc/../../.libs/tls/libgomp-7c85b1e2.so.1.0.0
     15433:       trying file=/tmp/ompwheel/numba/np/ufunc/../../.libs/x86_64/libgomp-7c85b1e2.so.1.0.0
     15433:       trying file=/tmp/ompwheel/numba/np/ufunc/../../.libs/libgomp-7c85b1e2.so.1.0.0
     15433:     calling init: /tmp/ompwheel/numba/np/ufunc/../../.libs/libgomp-7c85b1e2.so.1.0.0
     15433:     calling fini: /tmp/ompwheel/numba/np/ufunc/../../.libs/libgomp-7c85b1e2.so.1.0.0 [0]
```
seems fine, so I guess `patchelf` is now working correctly, thanks for re-enabling this @esc.

**Comment by esc** [comment]
@stuartarchibald thanks for checking!

**Pull request merged**